snapcraft/internal/build_providers/_base_provider.py

```diff
@@ -385,6 +385,13 @@ def _get_env_command(self) -> Sequence[str]:
             value = str(value)
             env_list.append(f"{key}={value}")
 
+        # Pass through build information environment variables.
+        for key in ["SNAPCRAFT_BUILD_INFO", "SNAPCRAFT_IMAGE_INFO"]:
```

Comment:
I think @cjp256 would agree, how about we make these self.build_provider_flags instead? That's in `snapcraft.cli._options._PROVIDER_OPTIONS`

Reply:
Yes, something like adding the following to provider options should be sufficient.
```
    dict(
        param_decls="--generate-manifest",
        is_flag=True,
        help="Generate snap manifest.",
        envvar="SNAPCRAFT_BUILD)INFO",
        supported_providers=["host", "lxd", "managed-host", "multipass"],
        hidden=True,
    ),
    dict(
        param_decls="--manifest-image-information",
        metavar="<json string>",
        help="Set snap manifest image-info",
        envvar="SNAPCRAFT_IMAGE_INFO",
        supported_providers=["host", "lxd", "managed-host", "multipass"],
        hidden=True,
    ),
```

Tests won't be required as it's covered with the generic provider option handling.

Reply:
I've updated the PR to make these options.

For compatibility with the existing call sites, I've updated `get_build_provider_flags` to not include `SNAPCRAFT_BUILD_INFO` if the option is turned off.  Without this, we'd set the environment variable to `"False"`, which be considered true by the various `if os.getenv("SNAPCRAFT_BUILD_INFO"):` calls.

Reply:
Good idea, works for the existing flags. Thanks!